### ISSUE TITLE
error when using config file

### ISSUE THREAD

**Issue by Tloe** [deepmap/oapi-codegen#1034]
I am running the generator with a config style and getting this error:
```
 ~/dev/pro/authly-core/in/api/openapi │ master  oapi-codegen -config config.yaml serviceapi.yaml
error parsing configuration style as old version or new version: <nil>
```

Looking at the code the logic seems a bit of? It also prints err which isn't the one it checks?
https://github.com/deepmap/oapi-codegen/blob/fd06f5aed35010af41fe12a17a0f75e16404af81/cmd/oapi-codegen/oapi-codegen.go#L156

**Comment by Festlandtommy** [comment]
I ran into the same issue, the config example in the readme has not been updated.
This is working for me:

```
package: api
generate:
  echo-server: true
  models: true
output: api.gen.go
```

Check these for the new config params:
https://github.com/deepmap/oapi-codegen/blob/fd06f5aed35010af41fe12a17a0f75e16404af81/pkg/codegen/configuration.go#L14-L33
(for me it was `server: true -> echo-server: true`)

**Comment by jamietanna** [comment]
@deepmap-marcinr this may be a good one to see if we can fix before the release

**Comment by deepmap-marcinr** [comment]
Man, updating to a new config file format and trying to auto-detect the format was sure a mistake on my part.

**Comment by deepmap-marcinr** [comment]
> 

@Tloe  - which version of oapi-codegen are you using? This seems to work ok on the latest commit.

**Comment by deitch** [comment]
I am glad I stumbled across this issue. I am getting the same `error parsing configuration style as old version or new version`, but it doesn't tell me what indicates which version it would be. How do I control that? And how do I know if my config is right for one version or the other?

> updating to a new config file format and trying to auto-detect the format was sure a mistake on my part

Heh, I sympathize. We all have been there. 

In my more recent projects, I have adopted a Kubernetes-style yaml, where a required top-level key is named `version` and another is `spec` or `config`, so all of the variants are inside there. It eliminates auto-detection.
